### DISCUSSION TITLE
Jhipster 8.5.0: Use podman instead of Docker with Testcontainers

### DISCUSSION THREAD

**Discussion by djibythiaw** [jhipster/generator-jhipster#26374] [Q&A]
Hello,
I want to use podman instead of Testcontainers while running my tests. Is it possible to do that ? If yes, how to configure it ?
Regards

**Comment by atomfrede** [comment]
You want to use podman instead of docker with testcontainers or would like to replace testcontainers?
If you want to use podman instead of docker (podman is a docker alternative) you should consult the[ testcontainers documentation for podman usage](https://java.testcontainers.org/supported_docker_environment/#podman). But I have never used podman myself, but as far as I heard from others its working fine.

**Reply by anyuruf** [reply]
on my fedora before I run the test, I run "systemctl --user start podman.socket" in the terminal.

**Comment by TatyOko28** [comment]
Yes. Testcontainers can use Podman as a Docker-compatible runtime.

Basic setup (Linux)
```bash
systemctl --user enable --now podman.socket
export DOCKER_HOST=unix://${XDG_RUNTIME_DIR}/podman/podman.sock
export TESTCONTAINERS_RYUK_DISABLED=true
```

Then run:

```bash
./mvnw test
```

Optional config (`~/.testcontainers.properties`):

```properties
docker.host=unix:///run/user/1000/podman/podman.sock
ryuk.container.privileged=false
```

Summary
Keep Testcontainers
 Use Podman socket instead of Docker
 Rootless Podman may need adjustments
